_posts/2023-01-04-trino-2022-the-rabbit-reflects.md

```diff
@@ -0,0 +1,581 @@
+The [launch of fault-tolerant execution
+]({% post_url 2022-05-05-tardigrade-launch %}) launch happened at
+Cinco de Trino. The first iterations only applied to queries being run on
+object-storage connectors such as Hive, Iceberg, and Delta Lake. Recently,
+support for MySQL, PostgreSQL, and SQL Server were added. These contributions
+added a foundation for other JDBC connectors.
+
+### SQL language improvements
+
+Here are all the notable SQL features that made it in to Trino this year:
+
+* [`MERGE` statement support]({{site.url}}/docs/current/sql/merge.html) is
+ the most impactful SQL feature released this year. `MERGE` allows users to
+ implement `INSERT`, `UPDATE`, and `DELETE` statements in one fell swoop.
+ `MERGE` is not simply syntax sugar, it actually has profound performance
+ implications as now a lot of your operations can be merged (pun intended) from
+ multiple tasks into a single scan over data. This functionality is absolutely
+ critical for positioning Trino as a data lakehouse query engine. `MERGE` is 
+ currently available in Hive, Iceberg, Delta Lake, Kudu, and Raptor connectors.
+ We discussed this and did a demo with `MERGE` on the recent [Trino Community
+ Broadcast with Iceberg]({{site.url}}/episodes/40.html).
+* Another massive update was the introduction of [polymorphic table functions
+ ]({% post_url 2022-07-22-polymorphic-table-functions %}) ([PTFs
+ ]({{site.url}}/docs/current/functions/table.html)). Table functions
+ initially released with some initial passthrough query functionality that we
+ see in connectors like Pinot, Elasticsearch, MySQL, PostgreSQL, and many
+ others. However, this is only one small instance of what can be achieved with
+ PTFs and the [true power comes from the generalization of this feature
+ ](https://www.youtube.com/clip/UgkxQcokpdgPjiuMKMC5-3HwHvlbmZjxAvxe). Dain and
+ David gave [a simpler explanation of PTFs
+ ](https://www.youtube.com/clip/Ugkx62IKgPd_v9eGBaPUHP2hyaRkWSXh8w8h) if it’s
+ still not very clear what they are. To dive in deeper, watch [this episode of
+ the Trino Community Broadcast]({{site.url}}/episodes/38.html) where Kasia
+ Findeisen and Martin discuss PTFs in greater detail.
+* [Dynamic function resolution](https://github.com/trinodb/trino/issues/8) has
+ been discussed for many years. This provides the ability for [connectors to
+ provide functions at runtime](https://youtu.be/mUq_h3oArp4?t=680). Unlike
+ before, where you needed to statically register your functions ahead of time,
+ you can now provide a plugin that defines these functions that gets resolves
+ while Trino is running. This enables features like supporting function calls to
+ dynamically registered user-defined functions in different languages like
+ Javascript or Python. Martin and Dain go into great detail about how this works
+ when [answering this question at Trino Summit
+ ](https://youtu.be/mUq_h3oArp4?t=1596).
+* [ANSI SQL 2016](https://en.wikipedia.org/wiki/SQL:2016) support for JSON 
```

Comment:
Added support for some JSON processing function, which are part of the ANSI 2016 specification.